src/Components/Context/ProductContext.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+
+export const ProductContext = createContext();
+
+export const ProductProvider = ({ children }) => {
+  const [apiProductData, setApiProductData] = useState([]);
+
+  useEffect(() => {
```

Comment:
I was doing it like that so that I need to call the API only once on page load and then pass this data on respective components.
Do correct me if I am thinking the wrong way.